### PR TITLE
wgengine/filter, tailcfg: support CIDRs+ranges in PacketFilter (mapver 7)

### PR DESCRIPTION
**Pull request by bradfitz**

### CODE REVIEW

File: tailcfg/tailcfg.go [hunk 1]
```diff
@@ -526,11 +527,11 @@ type PortRange struct {
 
 var PortRangeAny = PortRange{0, 65535}
 
-// NetPortRange represents a single subnet:portrange.
+// NetPortRange represents a range of ports that's allowed for one or more IPs.
 type NetPortRange struct {
 	_     structs.Incomparable
-	IP    string // "*" means all
-	Bits  *int   // backward compatibility: if missing, means "all" bits
+	IP    string // IP, CIDR, Range, or "*" (same formats as FilterRule.SrcIPs)
```
Comment (crawshaw) [inline tailcfg/tailcfg.go:533]: If I understand this correctly, we need to ensure that the server doesn't send a range if version < 7. Do we have any test in control for this? Any idea what it should look like?

File: wgengine/filter/filter_test.go [hunk 2]
```diff
@@ -560,3 +570,6 @@ func netports(netPorts ...string) (ret []NetPortRange) {
 	}
 	return ret
 }
+
+func TestMatchesFromFilterRules(t *testing.T) {
```
Comment (crawshaw) [inline wgengine/filter/filter_test.go:574]: oops?

**Review by crawshaw — APPROVED** [review]

File: tailcfg/tailcfg.go (hunk 1 above)
Comment (bradfitz) [reply tailcfg/tailcfg.go:533]: Yes, the mapCap stuff will be plumbed around. We don't need that until we start wanting to generate them.

File: wgengine/filter/filter_test.go (hunk 2 above)
Comment (bradfitz) [reply wgengine/filter/filter_test.go:574]: oh, yes, I started to add this then realized we already had tests for it.

Comment (bradfitz) [reply wgengine/filter/filter_test.go:574]: done.

**Pull request merged**